incubator/hnc/api/v1alpha1/hierarchy_types.go

```diff
@@ -70,8 +81,35 @@ type HierarchyList struct {
 	Items           []Hierarchy `json:"items"`
 }
 
+// Code is a machine-readable string value summarizing the condition.
+// +kubebuilder:validation:Enum=CRIT_PARENT_MISSING;CRIT_PARENT_INVALID;CRIT_REQUIRED_CHILD_CONFLICT;CRIT_ANCESTOR;OBJECT_OVERRIDDEN;OBJECT_DESCENDANT_OVERRIDDEN
+type Code string
+
+// Condition specifies the condition and the affected objects.
 type Condition struct {
-	Msg string `json:"msg"`
+	// Defines the conditions in a machine-readable string value.
+	// Valid values are:
+	// - "CRIT_PARENT_MISSING": the specified parent is missing;
```

Comment:
Nit: I'd drop the semicolon at the end of each line since these are all independent bullet points.

Reply:
done

Reply:
done